addons/website/controllers/main.py

```diff
@@ -117,6 +117,16 @@ def website_force(self, website_id, path='/', isredir=False, **kw):
         website._force()
         return request.redirect(path)
 
+    @http.route('/@/<path:path>', type='http', auth='public', website=True, sitemap=False, multilang=False)
+    def client_action_redirect(self, path, **kw):
+        path = f'/{path}?{request.httprequest.environ["QUERY_STRING"]}'
+        if request.env.user._is_internal():
+            path = '/web#' + werkzeug.urls.url_encode({
+                'action': 'website.website_preview',
+                'path': path,
+            })
+        return request.redirect(path)
```

Comment:
Actually I found something that does not work:

- Going to `127.0.0.1:8069/@/contactus` -> goes to the contact us page in the backend, good :+1: 
- Going to `127.0.0.1:8069/@/` (or `127.0.0.1:8069/@`) -> goes to a frontend-side 404 instead of the homepage in the backend :grimacing: (probably just need to define the `/@` controller in the route list ?)

(funny that both in your dev and my first testing we actually tried to hit F5 on different pages that the homepage :laughing:)

I think we should do more:
- This code here should call `get_client_action_url` IMO, to share the code
- All occurrences to `website.website_preview` (maybe there is only the link in the top-left mini-nav and next to the /my link) should be reviewed to use the `/@/` syntax, especially since I think it would allow to remove some url-encoding/decoding I had a hard time being sure they are right (there won't be anything to encode ever as it would be left to `url_encode` of this controller if I am not mistaken (I clearly could be mistaken, I am really lost in my encoding/decoding understanding with url).

Reply:
Forgot to mention: the point of adapting the URL with `website.website_preview` would also be able to handle the "enable_editor" parameter in the `/@/` controller, using `get_client_action_url`

Reply:
> funny that both in your dev and my first testing we actually tried to hit F5 on different pages that the homepage 

Indeed, good catch:
- Actually I made it worked, then replaced `/@<string:path>` by `<path:path>` converter (better and handling correctly multi slash url) and I didn't test the homepage again. Shame on me.
- Worst, I didn't write any test, double shame on me.

Anyway:
- I fixed the homepage case
- I used the `get_client_action_url` which is the correct approach, there should not be any direct call to `website.website_preview` python side :+1: 
- I handled the `enable_editor` case
- I added some tests (quite simple ones, testing page, controllers, multi slashes and query strings)

Everything should be good now